CODEOWNERS

```diff
@@ -0,0 +1,2 @@
+# Set Puppet's DIO team as the default code owner
+* @puppetlabs/dio
```

Comment:
@highb @briancain should you all be listed here too?

Reply:
I'm ok with being listed here :+1: